audit.sh

```diff
@@ -0,0 +1,406 @@
+#!/bin/bash
+# This script is designed to automate creating and verifying subrepos
+# Every git subrepo clone will be verified with a sha1sum to prove they are identical to the cloned repo
+# Every update/fetch for a sub repo will be presented with an output of the commit differences
+
+# Colours and Formating
+
+REPOURL_RAW="https://raw.githubusercontent.com/rackerlabs/useful-scripts/"
+
+# EF<letter> = End Formatting <option>
+UNDERLINE="\e[4m"
+EFU="\e[24m"
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+PURPLE='\033[0;35m'
+EFC='\033[0m'
+
+# Usage info
+usage() {              # Print script usage function
+    echo "Usage: $0 [options]"
+    echo "           -a, --add     <reponame> <url>   Specify name of new repo"
+    echo "           -u, --update  <reponame>         Update existing repo"
+    echo "           -c, --check   <reponame>         Check for updates on existing repo"
+    echo "           -v, --verify  <reponame> [local] Verifies local branch (not been pushed to git)"
+    echo ""
+    echo "           -h, --help                      Print help (usage)"
+    exit 1
+}
+
+# Gracefully close script with a message
+error_message() {
+    printf '%s\n' "$1" >&2
+    exit 1
+}
+
+# Make sure any changes are made to a sub branch. This ensured a PR is made and no direct
+# "git push" is made to the master branch
+check_local_branch() {
+    if ! [[ "$2" == "" ]]; then
+        update="-$2"
+    else
+        update="$2"
+    fi
+    branch_name=$(echo "$1" | sed 's/\///g')
+    answer=""
+    current_branch=$(git branch | grep \* | cut -d ' ' -f2)
+
+    if [[ "$current_branch" == "master" ]]; then
+        echo
+        echo -e "$RED""Error: You appear to be working on the master branch!$EFC"
+        echo "Updating/adding should be added to a sub branch."
+        echo
+        while [[ ! ( "$answer" =~ (y|Y|ye|Ye|yes|YES|Yes)$ ) ]]; do
```

Comment:
I think this could be simplified somewhat. At least the regex.
~~~shell
read -r -p "Are you sure? [y/N] " response
case "$response" in
    [yY][eE][sS]|[yY]) 
        do_something
        ;;
    *)
        do_something_else
        ;;
esac
~~~